deps/wazuh_testing/wazuh_testing/tools/sources/report_generator.py

```diff
@@ -0,0 +1,366 @@
+                if error_list_host:
+                    error_dict[type] += [{host_log['name']:  error_list_host}]
+        return error_dict
+
+    @staticmethod
+    def keep_alive_log_parser(log_files):
+        keep_alive_regex = '(\d{4}/\d{2}/\d{2} \d{2}:\d{2}:\d{2}) wazuh\-remoted.* reading ' + \
+            "'(.*)\|(.*)\|(.*)\|(.*)\|(.* \[.*\].*)\n(.*)\n.*:(\d+\.\d+\.\d+\.\d+)"
+
+        keep_alives = {}
+        for log_file in log_files:
+            regex = compile(rf"{keep_alive_regex}", MULTILINE)
+
+            with open(log_file['logs']['ossec.log']) as log:
+                for match in regex.finditer(log.read()):
+                    if match.group(3) not in keep_alives:
+                        keep_alives[match.group(3)] = {"n_keep_alive": 1, "max_difference": 0, "mean_difference": 0,
+                                                       "last_keep_alive": match.group(1),
+                                                       "first_keep_alive": match.group(1)}
+                    else:
+                        keep_alives[match.group(3)]["n_keep_alive"] += 1
+
+                        last_keep_alive_datetime = datetime.strptime(keep_alives[match.group(3)]["last_keep_alive"],
+                                                                     '%Y/%m/%d %H:%M:%S')
+                        recent_keep_alive_datetime = datetime.strptime(match.group(1), '%Y/%m/%d %H:%M:%S')
+
+                        if keep_alives[match.group(3)]["max_difference"] < \
+                                abs(recent_keep_alive_datetime - last_keep_alive_datetime).seconds:
+                            keep_alives[match.group(3)]["max_difference"] = \
+                                abs(recent_keep_alive_datetime - last_keep_alive_datetime).seconds
+
+                        keep_alives[match.group(3)]["mean_difference"] += \
+                            abs(recent_keep_alive_datetime - last_keep_alive_datetime).seconds
+
+                        keep_alives[match.group(3)]["last_keep_alive"] = match.group(1)
+
+            for key in keep_alives.keys():
+                keep_alives[key]["mean_difference"] = \
+                    keep_alives[key]["mean_difference"]/keep_alives[key]["n_keep_alive"]
+
+            last_timestamp = None
+            with open(log_file['logs']['ossec.log']) as log:
+                log_lines = log.readlines()
+                for line in reversed(log_lines):
+                    last_log_line = line
+                    timestamp = LogAnalyzer.get_log_timestamp(last_log_line)
+                    if timestamp:
+                        last_timestamp = timestamp
+                        break
+
+            for agent in keep_alives.keys():
+                last_keep_alive = datetime.strptime(keep_alives[agent]['last_keep_alive'], '%Y/%m/%d %H:%M:%S')
+                keep_alives[agent] = \
+                    {**keep_alives[agent], **{'remainder': abs(last_timestamp - last_keep_alive).seconds}}
```

Comment:
It is only analyzed the keep-alive received by the manager. So the only regex we should check is `Reading ...`. The variable should be named an agent because there is a dictionary for each agent.

Reply:
Done [7448bb7](https://github.com/wazuh/wazuh-qa/pull/2502/commits/7448bb7edc3e9506d01084be4e384b48ceda7a48)